lib/business/roleModule.mjs

```diff
@@ -0,0 +1,56 @@
+import BusinessBase from "./business-base.mjs";
+
+const permissionFields = [
+    'Permission1', 'Permission2', 'Permission3', 'Permission4',
+    'Permission5', 'Permission6', 'Permission7', 'Permission8',
+    'Permission9', 'Permission10'
+];
+
+class RoleModule {
+    // Method to handle any processing before saving the RoleModule data
+   static beforeSave = async ({ req }) => {
+        const { RoleId, ModuleId, RoleModuleId } = req.body;
+
+        // If the RoleModuleId is 0, check for mandatory fields (RoleId, ModuleId, etc.)
+        if (RoleModuleId === 0) {
+            if (!RoleId || !ModuleId) {
+                throw new Error('RoleId and ModuleId are required.');
+            }
+        }
+
+        // Remove Permission1, Permission2, etc., and combine them into the Permissions field
+
+
+        // Create a permissions binary string from the individual Permission fields
+        const permissionsBinary = permissionFields
+            .map(field => req.body[field] === '1' || req.body[field] === true ? '1' : '0') // Check if field is '1' or '0'
```

Comment:
you should add condition in parenthesis - 
(req.body[field] === '1' || req.body[field] === true) ? '1': '0'